tests/e2e/event_handler/test_header_serializer.py

```diff
@@ -122,20 +125,32 @@ def test_api_gateway_http_headers_serializer(apigw_http_endpoint):
 def test_lambda_function_url_headers_serializer(lambda_function_url_endpoint):
     # GIVEN
     url = f"{lambda_function_url_endpoint}todos"  # the function url endpoint already has the trailing /
+    body = "Hello World"
+    status_code = 200
+    headers = {"Content-Type": "text/plain", "Vary": ["Accept-Encoding", "User-Agent"]}
+    cookies = [
+        Cookie(name="session_id", value=str(uuid4()), secure=True, http_only=True),
+        Cookie(name="ab_experiment", value="3"),
+    ]
 
     # WHEN
-    response = data_fetcher.get_http_response(Request(method="GET", url=url))
+    response = data_fetcher.get_http_response(
+        Request(
+            method="POST",
+            url=url,
+            json={"body": body, "status_code": status_code, "headers": headers, "cookies": list(map(str, cookies))},
+        )
+    )
 
     # THEN
-    assert response.status_code == 200
-    assert response.content == b"Hello world"
-    assert response.headers["content-type"] == "text/plain"
+    assert response.status_code == status_code
+    assert response.content.decode("ascii") == body
 
```

Comment:
could you add a note here on the assertion?  it has quite a bit of logic we will forget when returning at some point.